packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,111 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import { DECIMAL_UNITS, bn } from 'fuels';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  balance: BN;
+  value: BN;
+  onChange: (val: BN) => void;
+};
+
+type AmountInputComponent = FC<AmountInputProps> & {
+  Loader: typeof AmountInputLoader;
+};
+
+export const AmountInput: AmountInputComponent = ({
+  balance,
+  value,
+  onChange,
+}) => {
+  const [assetAmount, setAssetAmount] = useState<string>(
+    value.eq(0) ? '' : value.toString()
+  );
+
+  const handleAmountChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newAssetAmount = handleAmountLeadingZeros(event);
+    onChange(bn.parseUnits(event.target.value, DECIMAL_UNITS));
+    setAssetAmount(newAssetAmount);
+  };
+
+  const handlePress = () => {
+    onChange(balance);
+    setAssetAmount(balance.format({ precision: DECIMAL_UNITS }));
+  };
+
+  const handleAmountLeadingZeros = (
+    event: ChangeEvent<HTMLInputElement>
+  ): string => {
+    const valueWithoutLeadingZeros = event.target.value.replace(
+      /^0\d/,
+      (substring) => substring.replace(/^0+(?=[\d])/, '')
+    );
+    return valueWithoutLeadingZeros.startsWith('.')
+      ? `0${valueWithoutLeadingZeros}`
+      : valueWithoutLeadingZeros;
+  };
+
+  return (
+    <Input size="lg" css={styles.input}>
+      <Input.Number
+        autoComplete="off"
+        inputMode="decimal"
+        name="amount"
+        placeholder="0.00"
+        allowedDecimalSeparators={['.', ',']}
+        allowNegative={false}
+        thousandSeparator={false}
+        value={assetAmount}
+        onChange={handleAmountChange}
```

Comment:
small logic change - comment 2/2

```suggestion
        onChange={(e) => handleAmountChange(e.target.value)}
```